### PR TITLE
Upload (daily) CI results to Hub

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Let's start to upload the results to a Hub dataset [hf-internal-testing/transformers_daily_ci](https://huggingface.co/datasets/hf-internal-testing/transformers_daily_ci/tree/main/2024-05-31) and use this dataset to build a visualization dashboard 🚀 !

So far I take the advantage of the daily CI workflow run - upload directly from there instead of fetching the finished workflow run afterward.

I didn't change any format in the raw reports (but processed in the workflow), but I believe it contains everything required to perform visualization/analysis.

[A dummy run](https://github.com/huggingface/transformers/actions/runs/9320789057) gives [hf-internal-testing/transformers_daily_ci](https://huggingface.co/datasets/hf-internal-testing/transformers_daily_ci/tree/main/2024-05-31)

### CODE REVIEW

File: .github/workflows/slack-report.yml [hunk 1]
```diff
@@ -19,6 +19,8 @@ on:
         required: true
         type: string
 
+env:
+  TRANSFORMERS_CI_RESULTS_UPLOAD_TOKEN: ${{ secrets.TRANSFORMERS_CI_RESULTS_UPLOAD_TOKEN }}
```
Comment (ydshieh) [inline .github/workflows/slack-report.yml:23]: This is a new fine grained token with write access to [hf-internal-testing/transformers_daily_ci](https://huggingface.co/datasets/hf-internal-testing/transformers_daily_ci)

File: utils/notification_service.py [hunk 2]
```diff
@@ -1154,12 +1157,25 @@ def prepare_reports(title, header, reports, to_truncate=True):
     if not os.path.isdir(os.path.join(os.getcwd(), f"ci_results_{job_name}")):
         os.makedirs(os.path.join(os.getcwd(), f"ci_results_{job_name}"))
 
+    target_workflow = "huggingface/transformers/.github/workflows/self-scheduled-caller.yml@refs/heads/main"
+    is_scheduled_ci_run = os.environ.get("CI_WORKFLOW_REF") == target_workflow
```
Comment (ydshieh) [inline utils/notification_service.py:1161]: Only upload the results for the runs triggered by the  scheduled event .

File: .github/workflows/slack-report.yml (hunk 1 above)
Comment (LysandreJik) [reply .github/workflows/slack-report.yml:23]: Cool!

**Review by LysandreJik — APPROVED** [review]
Awesome!

**Comment by ydshieh** [comment]
Thanks for the review. I will merge once I applied the same changes to

> utils/notification_service_quantization.py

(sorry, I forget this file quite frequently)

**Pull request merged**